ts/src/woofipro.ts

```diff
@@ -0,0 +1,2637 @@
+            if (stopLoss !== undefined) {
+                const stopLossPrice = this.safeNumber2 (stopLoss, 'triggerPrice', 'price', stopLoss);
+                const stopLossOrder = {
+                    'side': closeSide,
+                    'algo_type': 'TP_SL',
+                    'trigger_price': this.priceToPrecision (symbol, stopLossPrice),
+                    'type': 'LIMIT',
+                    'reduce_only': true,
+                };
+                outterOrder['child_orders'].push (stopLossOrder);
+            }
+            if (takeProfit !== undefined) {
+                const takeProfitPrice = this.safeNumber2 (takeProfit, 'triggerPrice', 'price', takeProfit);
+                const takeProfitOrder = {
+                    'side': closeSide,
+                    'algo_type': 'TP_SL',
+                    'trigger_price': this.priceToPrecision (symbol, takeProfitPrice),
+                    'type': 'LIMIT',
+                    'reduce_only': true,
+                };
+                outterOrder['child_orders'].push (takeProfitOrder);
+            }
+            request['child_orders'] = [ outterOrder ];
+        }
+        params = this.omit (params, [ 'clOrdID', 'clientOrderId', 'client_order_id', 'postOnly', 'timeInForce', 'stopPrice', 'triggerPrice', 'stopLoss', 'takeProfit', 'trailingPercent', 'trailingAmount', 'trailingTriggerPrice' ]);
+        let response = undefined;
+        if (isStop) {
+            response = await this.v1PrivatePostAlgoOrder (this.extend (request, params));
+            //
+            // {
+            //     "success": true,
+            //     "timestamp": 1702989203989,
+            //     "data": {
+            //       "order_id": 13,
+            //       "client_order_id": "testclientid",
+            //       "algo_type": "STOP",
+            //       "quantity": 100.12
+            //     }
+            // }
+            //
+        } else {
+            response = await this.v1PrivatePostOrder (this.extend (request, params));
+            //
+            // {
+            //     "success": true,
+            //     "timestamp": 1702989203989,
+            //     "data": {
+            //       "order_id": 13,
+            //       "client_order_id": "testclientid",
+            //       "order_type": "LIMIT",
+            //       "order_price": 100.12,
+            //       "order_quantity": 0.987654,
+            //       "order_amount": 0.8,
+            //       "error_message": "none"
+            //     }
+            // }
+            //
+        }
+        const data = this.safeDict (response, 'data');
+        data['timestamp'] = this.safeInteger (response, 'timestamp');
+        const order = this.parseOrder (data, market);
+        order['type'] = type;
+        return order;
+    }
+
+    async editOrder (id: string, symbol: string, type:OrderType, side: OrderSide, amount: Num = undefined, price: Num = undefined, params = {}) {
```

Comment:
for some reason can't edit a limit postOnly order

```
n woofipro editOrder "1088558009" "XRP/USDC:USDC" limit buy 21 0.45           
Debugger attached.
2024-05-03T14:51:42.688Z
Node.js: v18.18.0
CCXT v4.3.14
woofipro.editOrder (1088558009, XRP/USDC:USDC, limit, buy, 21, 0.45)
```

Reply:
BadRequest woofipro {"success":false,"code":"-1005","message":"You can not change other order info."}